core/src/main/java/cucumber/api/TestCase.java

```diff
@@ -0,0 +1,63 @@
+package cucumber.api;
+
+import cucumber.api.event.TestCaseFinished;
+import cucumber.api.event.TestCaseStarted;
+import cucumber.runner.EventBus;
+import cucumber.runtime.ScenarioImpl;
+import gherkin.events.PickleEvent;
+import gherkin.pickles.PickleLocation;
+import gherkin.pickles.PickleTag;
+
+import java.util.List;
+
+public class TestCase {
+    private final PickleEvent pickleEvent;
+    private final List<TestStep> testSteps;
+
+    public TestCase(List<TestStep> testSteps, PickleEvent pickleEvent) {
+        this.testSteps = testSteps;
+        this.pickleEvent = pickleEvent;
+    }
+
+    public void run(EventBus bus) {
+        boolean skipNextStep = false;
+        bus.send(new TestCaseStarted(bus.getTime(), this));
+        ScenarioImpl scenarioResult = new ScenarioImpl(bus, pickleEvent.pickle);
+        for (TestStep step : testSteps) {
+            Result stepResult = step.run(bus, pickleEvent.pickle.getLanguage(), scenarioResult, skipNextStep);
+            if (stepResult.getStatus() != Result.PASSED) {
```

Comment:
By the introduction of an enum for the result statuses in #1133, this kind of problem should be eliminated.